rebalance/rebalance.py

```diff
@@ -146,7 +146,10 @@ def rebalance(plugin, outgoing_scid, incoming_scid, msatoshi: Millisatoshi=None,
     plugin.log("Invoice payment_hash: %s" % payment_hash)
     success_msg = ""
     try:
-        excludes = [outgoing_scid + "/0", incoming_scid + "/0"]
+        excludes = []
+        mychannels = plugin.rpc.listchannels(source=my_node_id)['channels']
+        for channel in mychannels:
+            excludes += [channel['short_channel_id'] + '/0', channel['short_channel_id'] + '/1']
```

Comment:
Shouldn't you whitelist the channels that are adjacent to you and are to `incoming_node_id` and `outgoing_node_id`, otherwise you'll not find a route, or will you?

Reply:
@cdecker That what I was thinking, but I just tried and it works

Reply:
The old version did:

```
    excludes = [outgoing_scid + "/0", incoming_scid + "/0"]
```

Which also worked but lead to the issue #17 . Also the usage of `"/0"` seemed wrong to me, but the new code simply blacklists all own channels. I think the c-lightning router removes these, otherwise it would not work.

Reply:
`getroute` would not return a route if the source and target node id are the same. I cannot get a circular route by using `getroute` only.
Instead, I do some workaround.
I set up the first and the last hop without `getroute`. The routes to my adjacent nodes are `route_out` and `route_in`.
https://github.com/lightningd/plugins/blob/4fdaaa461d49b5aff1f0e898beb65ff04197c2a8/rebalance/rebalance.py#L154
I use `getroute` to find a `route_mid` between `incoming_node_id` and `outgoing_node_id`. That's why I exclude all own channels.

Reply:
In this case adding all own channels in this exclude list seems correct

Reply:
@cdecker can we resolve this?